platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/ProjectConfigImpl.java

```diff
@@ -47,36 +156,83 @@ public SourceStorage getSource() {
         return storage;
     }
 
-    public void setStorage(SourceStorageImpl sourceStorage) {
+    @Override
+    public List<Link> getLinks() {
+        if (links == null) {
+            links = new ArrayList<>();
+        }
+        return links;
+    }
+
+    @Override
+    public String getContentRoot() {
+        return contentRoot;
+    }
+
+    public void setContentRoot(String contentRoot) {
+        this.contentRoot = contentRoot;
+    }
+
+    public void setLinks(List<Link> links) {
+        this.links = links;
+    }
+
+    public void setSource(SourceStorageImpl sourceStorage) {
         this.storage = sourceStorage;
     }
 
+    public void setProblems(List<ProjectProblem> problems) {
+        this.problems = problems;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (!(o instanceof ProjectConfigImpl)) return false;
-        if (!super.equals(o)) return false;
         final ProjectConfigImpl other = (ProjectConfigImpl)o;
-        return Objects.equals(storage, other.storage);
+        return Objects.equals(name, other.name) &&
```

Comment:
'&&' must go in a new line see [doc](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-any-other-operator-the-break-comes-before-the-symbol)